providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/process/targeting/FractionalTest.java

```diff
@@ -0,0 +1,220 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.targeting;
+
+import io.github.jamsesso.jsonlogic.evaluator.JsonLogicEvaluationException;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static dev.openfeature.contrib.providers.flagd.resolver.process.targeting.Operator.FLAG_KEY;
+import static dev.openfeature.contrib.providers.flagd.resolver.process.targeting.Operator.TARGET_KEY;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class FractionalTest {
+
+    @Test
+    void selfContainedFractional() throws JsonLogicEvaluationException {
+        // given
+        Fractional fractional = new Fractional();
+
+        /* Rule
+         *     [
+         *       "bucketKey",
+         *       [
+         *         "red",
+         *         50
+         *       ],
+         *       [
+         *         "blue",
+         *         50
+         *       ]
+         *     ]
+         * */
+
+        final List<Object> rule = new ArrayList<>();
+        rule.add("bucketKey");
+
+        final List<Object> bucket1 = new ArrayList<>();
+        bucket1.add("red");
+        bucket1.add(50);
+
+        final List<Object> bucket2 = new ArrayList<>();
+        bucket2.add("green");
+        bucket2.add(50);
+
+        rule.add(bucket1);
+        rule.add(bucket2);
+
+        Map<String, String> data = new HashMap<>();
+        data.put(FLAG_KEY, "flagA");
```

Comment:
I'm confused by this test, it doesn't set a value for `bucketKey` in data... I would expect to see that have different values, and see differential results when it does.

Reply:
This is related to this comment - https://github.com/open-feature/java-sdk-contrib/pull/434#discussion_r1324915298

By the time our implementation receives arguments, JsonLogic expressions will be resolved to their values. The test here only validate the rule it receive (bucketing key set to `bucketKey` and flag key set to `flagA`).

Hope this helps to clarify your doubt ? 

Note - I think I need to add more tests to cover targeting rules as a complete unit (or component) tests

Reply:
> Note - I think I need to add more tests to cover targeting rules as a complete unit (or component) tests

Honestly, no, I think if we're on the same page, I will just implement the gherkin suite which should test those things.

Reply:
I think, if anything, you'd want to run this test twice... once with "bucketKeyA" and once with "bucketKeyB" or some other value that resolves in a different random assignment. Does that make sense?

Reply:
this is done :) and there's variation validation thanks to this.